File: backend-C#/Controllers/SocialController.cs
using IMS.DTOs;
using IMS.DTOs.Requests;
using IMS.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace IMS.Controllers;

[ApiController]
[Route("api/social")]
[Authorize]
public class SocialController : BaseController
{
    private readonly ISocialService _socialService;

    public SocialController(ISocialService socialService)
    {
        _socialService = socialService;
    }

    // Posts
    [HttpGet("posts")]
    public async Task<ActionResult<ApiResponse>> GetPosts([FromQuery] ListPostsRequest request)
    {
        if (!CurrentUserId.HasValue)
        {
            return Error("未登录", 401);
        }

        var result = await _socialService.GetPostsAsync(CurrentUserId.Value, request);
        return Success(new 
        { 
            posts = result.Posts, 
            total = result.Total, 
            page = result.Page, 
            limit = result.Limit 
        }, "获取成功");
    }

    [HttpPost("posts")]
    public async Task<ActionResult<ApiResponse>> CreatePost([FromBody] CreatePostRequest request)
    {
        if (!CurrentUserId.HasValue)
        {
            return Error("未登录", 401);
        }

        try
        {
            var post = await _socialService.CreatePostAsync(CurrentUserId.Value, request);
            return Success(post, "创建成功");
        }
        catch (InvalidOperationException ex)
        {
            return Error(ex.Message, 400);
        }
    }

    [HttpGet("posts/{id}")]
    public async Task<ActionResult<ApiResponse>> GetPost(long id)
    {
        // Return empty response for now - can be extended
        return Success(null, "获取成功");
    }

    [HttpPut("posts/{id}")]
    public async Task<ActionResult<ApiResponse>> UpdatePost(long id)
    {
        // Placeholder
        return Success(new { }, "更新成功");
    }

    [HttpDelete("posts/{id}")]
    public async Task<ActionResult<ApiResponse>> DeletePost(long id)
    {
        if (!CurrentUserId.HasValue)
        {
            return Error("未登录", 401);
        }

        try
        {
            await _socialService.DeletePostAsync(CurrentUserId.Value, CurrentUserRole ?? "", id);
            return Success("删除成功");
        }
        catch (InvalidOperationException ex)
        {
            return Error(ex.Message, 400);
        }
    }

    [HttpPost("posts/{id}/like")]
    public async Task<ActionResult<ApiResponse>> ToggleLike(long id)
    {
        if (!CurrentUserId.HasValue)
        {
            return Error("未登录", 401);
        }

        try
        {
            var result = await _socialService.ToggleLikeAsync(CurrentUserId.Value, id);
            return Success(result, "操作成功");
        }
        catch (InvalidOperationException ex)
        {
            return Error(ex.Message, 400);
        }
    }

    [HttpPost("posts/{id}/comment")]
    public async Task<ActionResult<ApiResponse>> AddComment(long id, [FromBody] AddCommentRequest request)
    {
        if (!CurrentUserId.HasValue)
        {
            return Error("未登录", 401);
        }

        try
        {
            var comment = await _socialService.AddCommentAsync(CurrentUserId.Value, id, request);
            return Success(comment, "评论成功");
        }
        catch (InvalidOperationException ex)
        {
            return Error(ex.Message, 400);
        }
    }

    [HttpGet("posts/{id}/comments")]
    public async Task<ActionResult<ApiResponse>> GetComments(long id, [FromQuery] int page = 1, [FromQuery] int limit = 20)
    {
        var result = await _socialService.GetCommentsAsync(id, page, limit);
        return Success(new 
        { 
            comments = result.Comments, 
            total = result.Total, 
            page = result.Page, 
            limit = result.Limit 
        }, "获取成功");
    }
}

File: backend-JS/src/controllers/socialController.js
const { socialService } = require('../services');
const { success, created, error } = require('../utils/response');

const getPosts = async (req, res) => {
  try {
    const page = parseInt(req.query.page) || 1;
    const limit = parseInt(req.query.limit) || 10;
    
    const result = await socialService.getPosts(req.userId, page, limit);
    return success(res, result, '获取成功');
  } catch (err) {
    return error(res, 500, '获取帖子失败');
  }
};

const createPost = async (req, res) => {
  try {
    const post = await socialService.createPost(req.userId, req.body);
    return created(res, post, '创建成功');
  } catch (err) {
    return error(res, 400, err.message);
  }
};

const deletePost = async (req, res) => {
  try {
    await socialService.deletePost(parseInt(req.params.id), req.userId);
    return success(res, null, '帖子已删除');
  } catch (err) {
    return error(res, 400, err.message);
  }
};

const toggleLike = async (req, res) => {
  try {
    const result = await socialService.toggleLike(parseInt(req.params.id), req.userId);
    return success(res, result, result.liked ? '点赞成功' : '取消点赞');
  } catch (err) {
    return error(res, 400, err.message);
  }
};

const addComment = async (req, res) => {
  try {
    const comment = await socialService.addComment(parseInt(req.params.id), req.userId, req.body);
    return created(res, comment, '评论成功');
  } catch (err) {
    return error(res, 400, err.message);
  }
};

const testPosts = async (req, res) => {
  try {
    const { Post } = require('../models');
    const posts = await Post.findAll({ order: [['createdAt', 'DESC']] });
    return success(res, { posts, count: posts.length }, '获取成功');
  } catch (err) {
    return error(res, 500, '测试失败');
  }
};

module.exports = {
  getPosts,
  createPost,
  deletePost,
  toggleLike,
  addComment,
  testPosts
};
